internal/cmd/beta/server/create/create.go

```diff
@@ -0,0 +1,338 @@
+			),
+			examples.NewExample(
+				`Create a server with labels from an image with id xxx`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --image-id xxx --labels key=value,foo=bar`,
+			),
+			examples.NewExample(
+				`Create a server with a boot volume`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --boot-volume-source-id xxx --boot-volume-source-type image --boot-volume-size 64`,
+			),
+			examples.NewExample(
+				`Create a server with a boot volume from an existing volume`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --boot-volume-source-id xxx --boot-volume-source-type volume`,
+			),
+			examples.NewExample(
+				`Create a server with a keypair`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --image-id xxx --keypair-name example`,
+			),
+			examples.NewExample(
+				`Create a server with a network`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --image-id xxx --network-id yyy`,
+			),
+			examples.NewExample(
+				`Create a server with a network interface`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --boot-volume-source-id xxx --boot-volume-source-type image --boot-volume-size 64 --network-interface-ids yyy`,
+			),
+			examples.NewExample(
+				`Create a server with an attached volume`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --boot-volume-source-id xxx --boot-volume-source-type image --boot-volume-size 64 --volumes yyy`,
+			),
+			examples.NewExample(
+				`Create a server with user data (cloud-init)`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --boot-volume-source-id xxx --boot-volume-source-type image --boot-volume-size 64 --user-data file("${path.module}/cloud-init.yaml")`,
+			),
+		),
+		PreRun: func(cmd *cobra.Command, _ []string) {
+			bootVolumeSourceId, _ := cmd.Flags().GetString(bootVolumeSourceIdFlag)
+			bootVolumeSourceType, _ := cmd.Flags().GetString(bootVolumeSourceTypeFlag)
+			bootVolumeSize, _ := cmd.Flags().GetInt64(bootVolumeSizeFlag)
+			imageId, _ := cmd.Flags().GetString(imageIdFlag)
+
+			if imageId == "" && bootVolumeSourceId == "" && bootVolumeSourceType == "" {
+				p.Warn("Either Image ID or boot volume flags must be provided.\n")
+			}
+
+			if imageId == "" {
+				err := flags.MarkFlagsRequired(cmd, bootVolumeSourceIdFlag, bootVolumeSourceTypeFlag)
+				cobra.CheckErr(err)
+			}
+
+			if bootVolumeSourceType == "image" {
+				if bootVolumeSize == 0 {
+					p.Warn("Boot volume size must be provided when `source_type` is `image`.\n")
+				}
+				err := cmd.MarkFlagRequired(bootVolumeSizeFlag)
+				cobra.CheckErr(err)
+			}
+
+			if bootVolumeSourceId == "" && bootVolumeSourceType == "" {
+				err := cmd.MarkFlagRequired(imageIdFlag)
+				cobra.CheckErr(err)
+			}
+
+			if imageId != "" && (bootVolumeSourceId != "" || bootVolumeSourceType != "") {
```

Comment:
For this we also already use some utils e.g. `MarkFlagsMutuallyExclusive` in the `configureFlags`